bin/img-set.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env node
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+const fs = require('fs');
+const path = require('node:path');
+const im = require('node-imagemagick');
+let imgFile = process.argv[2]; // get argument passed to script
+let sizes = [300, 500, 700, 900, 1100, 1300, 1500]; // what sizes do we want?
```

Comment:
Possible issue / suggestion: I'm a little wary about hard coding a bunch of set image sizes like this, as developers might just copy and paste the results here without checking if the sizes are actually suitable for the image being displayed?

This is only a rough example, but when doing this manually my approach would be to:

1. Work out the default, largest display width for an image on the page in px (e.g. 500px).
2. Double that value for high-res screens (e.g. 1000px).
3. Then make variations from that size downward by 20-25% (e.g. 1000px, 750px, 500px, 250px).

Perhaps we could work out the width of the image passed in (or alternatively specify the width on the command line), and the use that as a baseline for the sizes to generate. Thoughts?

Reply:
Yeah, I didn't like hard coding them either but I didn't have a formula for deciding how to make them on the fly and I opted to submit what I had rather than keep staring at them. I do like the idea of taking the largest needed size as a parameter and doing math from there, since it forces the dev to think about it. It'd be nice if they didn't have to think about it for standard Protocol components though.

I've been reading up on the general advice for deciding how many and what sizes and the advice that stuck out the most to me is to make steps based on the file size rather than the image size. Which would be pretty complex to do on the fly. There is an [open source tool and API for it ](https://www.responsivebreakpoints.com/)though.

Accepting a parameter and doing fixed sizes from there feels like the best sort-term solution.

Reply:
I'm abdicating all responsibility and just asking for a full array of sizes. It's not a time saver from a math perspective  but it is still a time saver for the resizing.